easybuild/framework/easyblock.py

```diff
@@ -3350,6 +3352,271 @@ def _sanity_check_step_multi_deps(self, *args, **kwargs):
+                            # - Missing device code is considered a failure (unless there is PTX code for
+                            #   a lower CC AND --accept-ptx-for-cc-support is True, in which case it is a warning)
+                            # - Device code for additional compute capabilities is considered a failure if
+                            #   --strict-cuda-sanity-check is True (otherwise, it's a warning)
+                            # - Missing PTX code for the highest CUDA compute capability in --cuda-compute-capabilities
+                            #   is considered a failure, unless --accept-missing-cuda-ptx is True (in which case it is
+                            #   a warning)
+                            num_cuda_files += 1
+                            # unpack results
+                            derived_ccs = res.device_code_archs
+                            derived_ptx_ccs = res.ptx_archs
+
+                            # check whether device code architectures match cuda_compute_capabilities
+                            additional_ccs = list(set(derived_ccs) - set(cfg_ccs))
+                            missing_ccs = list(set(cfg_ccs) - set(derived_ccs))
+
+                            # Message for when file is on the ignore list:
+                            ignore_msg = f"This failure will be ignored as '{path}' is listed in "
+                            ignore_msg += "'ignore_cuda_sanity_failures'."
+
+                            if additional_ccs or missing_ccs:
+                                fail_msg = f"Mismatch between cuda_compute_capabilities and device code in {path}. "
+                                if additional_ccs:
+                                    # Count and log for summary report
+                                    files_surplus_cc.append(path)
+                                    num_files_surplus_cc += 1
+                                    surplus_cc_str = ', '.join(sorted(additional_ccs, key=LooseVersion))
+                                    fail_msg += "Surplus compute capabilities: %s. " % surplus_cc_str
+                                    if strict_cc_check:  # Surplus compute capabilities not allowed
+                                        if path in ignore_file_list or ignore_failures:
+                                            files_surplus_cc_ignored.append(path)
+                                            num_files_surplus_cc_ignored += 1
+                                            fail_msg += ignore_msg
+                                            is_failure = False
+                                        else:
+                                            files_surplus_cc_fails.append(path)
+                                            num_files_surplus_cc_fails += 1
+                                            is_failure = True
+                                    else:
+                                        is_failure = False
+
```

Comment:
add log info (or warning) message here to show how many failing files were found:
```suggestion
self.log.info(f"{len(files_surplus_cc) files found with additional device code architecture: {files_surplus_cc}")
```

maybe only paths relative to install dir, something like:

```python
[os.path.relpath(x, self.installdir) for x in files_surplus_cc]

Reply:
I'll actually do this when _appending_ the path, so that the long path never makes it into the list. No point, really.

Reply:
Wait, no, you're suggested location actually doesn't make sense: this is _inside_ the loop. That would print an increasingly growing list.

We can do two things:
- print the _individual_ path to `self.log.info` right away (i.e. at the location where we `append` it to some list)
- print the lists right after the for-loop

Now that we've reduce the length of the paths, I'm in favor of doing this after the for loop (but then before the 'summary in numbers', so that that's the last thing you get from the sanity check - that way it stands out).

Reply:
Ok, I'm now printing all, but after the for-loop.